main.nf

```diff
@@ -176,57 +248,479 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
+  process star {
+      label 'high_memory'
+
+      tag "$prefix"
+      publishDir "${params.outdir}/STAR", mode: 'copy'
+
+      when:
+      params.aligner == "star"
+
+      input:
+      set val(samplename), file(reads) from read_files_star
+      file index from star_index.collect()
+      file gtf from gtf_star.collect()
+      file whitelist from barcode_whitelist_star.collect()
+
+      output:
+      set file("*Log.final.out"), file ('*.bam') into star_aligned
+      file "*.out" into alignment_logs
+      file "*SJ.out.tab"
+      file "*Log.out" into star_log
+      file "${prefix}Aligned.sortedByCoord.out.bam.bai" into bam_index_rseqc, bam_index_genebody
+
+      script:
+      prefix = reads[0].toString() - ~/(_R1)?(_trimmed)?(_val_1)?(\.fq)?(\.fastq)?(\.gz)?$/
+      def star_mem = task.memory ?: params.star_memory ?: false
+      def avail_mem = star_mem ? "--limitBAMsortRAM ${star_mem.toBytes() - 100000000}" : ''
+
+      seqCenter = params.seqCenter ? "--outSAMattrRGline ID:$prefix 'CN:$params.seqCenter'" : ''
+      cdna_read = reads[0]
+      barcode_read = reads[1]
+      """
+      STAR --genomeDir $index \\
+           --sjdbGTFfile $gtf \\
+           --readFilesIn $barcode_read $cdna_read  \\
+           --runThreadN ${task.cpus} \\
+           --twopassMode Basic \\
+           --outWigType bedGraph \\
+           --outSAMtype BAM SortedByCoordinate $avail_mem \\
+           --readFilesCommand zcat \\
+           --runDirPerm All_RWX \\
+           --outFileNamePrefix $prefix $seqCenter \\
+           --soloType Droplet \\
+           --soloCBwhitelist $whitelist
+
+      samtools index ${prefix}Aligned.sortedByCoord.out.bam
+      """
+
+
+  }
+  // Filter removes all 'aligned' channels that fail the check
+  star_aligned
+      .filter { logs, bams -> check_log(logs) }
+      .flatMap {  logs, bams -> bams }
+  .into { bam_count; bam_rseqc; bam_preseq; bam_markduplicates; bam_htseqcount; bam_stringtieFPKM; bam_for_genebody; bam_dexseq; leafcutter_bam }
+} else {
+  star_log = Channel.empty()
+}
+// Run Kallisto bus
+
+if (params.aligner == 'kallisto'){
+  process kallisto {
+    tag "$name"
+    label 'mid_memory'
+    publishDir "${params.outdir}/kallisto/raw_bus", mode: 'copy'
+
+    when:
+    params.aligner == "kallisto"
+
+    input:
+    set val(name), file(reads) from read_files_kallisto
+    file index from kallisto_index.collect()
+
+    output:
+    file "${name}_bus_output" into kallisto_bus_to_sort
+    file "${name}_kallisto.log" into kallisto_log_for_multiqc
+
+    script:
+    """
+    kallisto bus \\
+        -i $index \\
+        -o ${name}_bus_output/ \\
+        -x ${params.type}${params.chemistry} \\
+        -t ${task.cpus} \\
+        $reads | tee ${name}_kallisto.log
+    """
+  }
+
+  process bustools_correct_sort{
+    tag "$bus"
+    label 'mid_memory'
+    publishDir "${params.outdir}/kallisto/sort_bus", mode: 'copy'
+
+    when:
+    params.aligner == "kallisto" && !params.skip_bustools
```

Comment:
You should just need `!params.skip_bustools` in your `when` statement, since channel `kallisto_bus_to_sort` is chained from process `kallisto` it depends from the previous process, but it should be after output.